testing/trino-testing/src/main/java/io/trino/testing/AbstractTestFailureRecovery.java

```diff
@@ -105,30 +128,54 @@ public void testAggregation()
     @Test(invocationCount = INVOCATION_COUNT)
     public void testJoin()
     {
-        testSelect("SELECT * FROM orders o, customer c WHERE o.custkey = c.custkey AND c.nationKey = 1");
+        @Language("SQL") String selectQuery = "SELECT * FROM partitioned_lineitem JOIN supplier ON partitioned_lineitem.suppkey = supplier.suppkey " +
+                "AND supplier.name = 'Supplier#000000001'";
+        testSelect(selectQuery, Optional.of(enableDynamicFiltering(false)), queryId -> {});
```

Comment:
Add a comment why do we test this case (e.g: `assert that query doesn't fail and return correct results when DF is off` 

nit: assert number of rows maybe? Otherwise you can just use `testSelect` without assertion overload.